docs/tasks/administer-cluster/encrypt-data.md

```diff
@@ -0,0 +1,205 @@
+
+{% endcapture %}
+
+{% capture steps %}
+
+## Configuration and determining whether encryption at rest is already enabled
+
+The `kube-apiserver` process accepts an argument `--experimental-encryption-provider-config`
+that controls how API data is encrypted in etcd. An example configuration
+is provided below.
+
+## Understanding the encryption at rest configuration.
+
+```yaml
+kind: EncryptionConfig
+apiVersion: v1
+resources:
+  - resources:
+    - secrets
+    providers:
+    - identity: {}
+    - aesgcm:
+        keys:
+        - name: key1
+          secret: c2VjcmV0IGlzIHNlY3VyZQ==
+        - name: key2
+          secret: dGhpcyBpcyBwYXNzd29yZA==
+    - aescbc:
+        keys:
+        - name: key1
+          secret: c2VjcmV0IGlzIHNlY3VyZQ==
+        - name: key2
+          secret: dGhpcyBpcyBwYXNzd29yZA==
+    - secretbox:
+        keys:
+        - name: key1
+          secret: YWJjZGVmZ2hpamtsbW5vcHFyc3R1dnd4eXoxMjM0NTY=
+```
+
+Each `resources` array item is a separate config and contains a complete configuration. The
+`resources.resources` field is an array of Kubernetes resource names (`resource` or `resource.group`)
+that should be encrypted. The `providers` array is an ordered list of the possible encryption
+providers. Only one provider type may be specified per entry (`identity` or `aescbc` may be provided,
+but not both in the same item).
+
+The first provider in the list will be used to encrypt resources going into storage. When reading
+resources from storage the providers will be attempted in order to decrypt data. If no provider
+can read the stored data, an error will be returned which will prevent clients from accessing that
+resource. 
+
+IMPORTANT: If any resource is not readable via the encryption config (because keys were changed), 
+the only recourse is to delete that key from the underlying etcd directly. Calls that attempt to 
+read that resource will fail until it is deleted or a valid decryption key is provided.
+
+### Providers:
+
+* `identity` results in the data being written as-is without encryption. 
+  * When placed in the first position the resource will be decrypted as new values are written.
+* `secretbox` uses XSalsa20 and Poly1305 to store data at rest
+  * It is fast, but a newer standard and may not be considered acceptable in environments that require high levels of review. 
+  * It requires a 32 byte key.
+* `aescbc` uses AES in CBC mode
+  * It is the recommended choice for encryption at rest but may be slightly slower than `secretbox`.
+  * It requires 32 byte keys
+* `aesgcm` uses AES in GCM mode with a randomly assigned nonce 
+  * This mode is the fastest, but is not recommended for use except when an automated key rotation scheme is implemented. * Because these nonces are small the secret key must be rotated frequently - at most every 200k writes.
```

Comment:
Were these 2 different bullet points which got merged into one? There seems to be a wayward `*` in the middle. Also, the sentence after that ampersand does not seem like a standalone sentence, because it starts with `Because`. Something like:
```
* Since these nonces are small, the secret key must be rotated frequently.
```